src/dynamodb.js

```diff
@@ -26,21 +26,41 @@ function buildConfigFromConnection (connection) {
 }
 
 async function createTables (dynamoClient, uniqueIdentifier) {
-  await Promise.all(tablesSchema.map(async table => {
-    const newTable = cloneDeep(table);
-    const TableName = getTableName(newTable.TableName, uniqueIdentifier);
-    newTable.TableName = TableName;
-    await dynamoClient.createTable(newTable).promise();
-    await dynamoClient.waitFor('tableExists', { TableName });
-  }));
+  const provisionedTables = [];
+  await Promise.all(
+    tablesSchema.map(async table => {
+      const newTable = cloneDeep(table);
+      const TableName = getTableName(newTable.TableName, uniqueIdentifier);
+      newTable.TableName = TableName;
+
+      try {
+        await dynamoClient.createTable(newTable).promise();
+        await dynamoClient.waitFor('tableExists', { TableName }).promise();
+        provisionedTables.push(TableName);
+      } catch (err) {
+        console.error(`Failed to create table "${TableName}"`, err);
+      }
+    })
+  );
+  return provisionedTables;
 }
 
-async function destroyTables (dynamoClient, uniqueIdentifier) {
-  await Promise.all(tablesSchema.map(async ({TableName}) => {
-    const fullTableName = getTableName(TableName, uniqueIdentifier);
-    await dynamoClient.deleteTable({ TableName: fullTableName }).promise();
-    await dynamoClient.waitFor('tableNotExists', { TableName: fullTableName });
-  }));
+async function destroyTables (dynamoClient, provisionedTables) {
+  const { TableNames } = await dynamoClient.listTables().promise();
+  const tablesToDestroy = TableNames
+    .filter(name => provisionedTables.includes(name));
+
+  await Promise.all(
+    tablesToDestroy
+      .map(async TableName => {
+        try {
+          await dynamoClient.deleteTable({ TableName }).promise();
+          await dynamoClient.waitFor('tableNotExists', { TableName }).promise();
+        } catch (err) {
+          console.error(`Failed to destroy table "${TableName}"`, err);
```

Comment:
How will you report the failure to destroy table back to the caller?

Reply:
Added a net error to be thrown after all attempts

Reply:
I removed the `provisionedTables` context. It just added mess, `listTables` should filter tables that weren't successfully provisioned anyways.